### PR TITLE
Ben/cleanup unused from finetune

### PR DESCRIPTION
**Pull request by benleetownsend**

### CODE REVIEW

File: finetune/base_models/gpt/featurizer.py [hunk 1]
```diff
@@ -231,6 +208,8 @@ def add_explain_tokens(X, max_length, pool_idx):
     clf_tok_x_seq_w_pos = tf.concat((clf_tok_x_seq, flat_pos), -1)
     return tf.concat((X, clf_tok_x_seq_w_pos), 1)
 
+def get_pos_values(seq_len, vocab_size):
```
Comment (madisonmay) [inline finetune/base_models/gpt/featurizer.py:211]: This is the change that concerns me the most because of all the little oddities w/ off by one errors, special tokens, mask tokens, etc. that impact where pos embeddings start.  Also the Roberta starting pos embeddings at 2 thing.

Comment (benleetownsend) [reply finetune/base_models/gpt/featurizer.py:211]: Roberta already handles it's own, we never used the input idx for roberta, just GPT and GPT2

Comment (benleetownsend) [reply finetune/base_models/gpt/featurizer.py:211]: When I printed them out, it seemed to be equivalent to this range call, can you think of any specific cases worth testing.

File: finetune/nn/target_blocks.py [hunk 2]
```diff
@@ -192,12 +191,6 @@ def classifier(hidden, targets, n_targets, config, train=False, reuse=None, **kw
                 clf_losses, targets, kwargs.get("class_weights")
             )
 
-            # From Unsupervised Data Augmentation for Consistency Training, Xie et al. 2019
-            if config.tsa_schedule:
```
Comment (lhz1029) [inline finetune/nn/target_blocks.py:196]: What were the results of the benchmarks here, out of curiosity? No substantial lift?

File: finetune/saver.py [hunk 3]
```diff
@@ -336,7 +253,7 @@ def remove_unchanged(self, variable_names, variable_values, fallback_vars):
                 if fb_var_name == var_name:
                     for func in self.variable_transforms:
                         fb_var = func(var_name, fb_var)
-                    if np.allclose(fb_var, var_val):
+                    if fb_var.shape == var_val.shape and np.allclose(fb_var, var_val):
```
Comment (lhz1029) [inline finetune/saver.py:263]: when might we get into a state where these shapes don't match?

File: finetune/model.py [hunk 4]
```diff
@@ -71,10 +71,9 @@ def language_model_op(X, M, params, featurizer_state, mode, encoder):
     return lm_predict_op, language_model_state
 
 
-def masked_language_model_op(X, M, mlm_weights, mlm_ids, mlm_positions, params, featurizer_state, mode):
+def masked_language_model_op(X, mlm_weights, mlm_ids, mlm_positions, params, featurizer_state, mode):
```
Comment (lhz1029) [inline finetune/model.py:74]: nice call with mlm_mask. unless I missed something, it seems like we also don't need mlm_weights either?

**Review by lhz1029 — APPROVED** [review]

File: finetune/base_models/gpt/featurizer.py (hunk 1 above)
Comment (madisonmay) [reply finetune/base_models/gpt/featurizer.py:211]: We have tests for this so ultimately not concerned, this was just the only thing that jumped out as having the potential to break something.

File: finetune/model.py (hunk 4 above)
Comment (madisonmay) [reply finetune/model.py:74]: Yeah I suppose that could be computed in graph with a scatter using mlm_positions?

File: finetune/model.py [hunk 5]
```diff
@@ -320,7 +295,7 @@ def optimizer(lr):
                 increment_global_step=True,
                 summaries=summaries,
                 colocate_gradients_with_ops=True,
-                variables=params.trained_variables,
+#                variables=trained_variables,
```
Comment (madisonmay) [inline finetune/model.py]: Yea seems like we can cut this out entirely.  Default seems to be train all trainable params which is what we did explicitly before.

File: finetune/input_pipeline.py [hunk 6]
```diff
@@ -122,49 +123,29 @@ def _array_format(self, encoded_output):
             1: positional embedding
         """
         seq_length = len(encoded_output.token_ids)
-        x = np.zeros((seq_length, 2), dtype=np.int32)
+        x = np.zeros((seq_length), dtype=np.int32)
```
Comment (madisonmay) [inline finetune/input_pipeline.py]: we probably don't need to allocate this array of zeros anymore.  could just say x = `encoded_output.token_ids`

File: finetune/saver.py [hunk 7]
```diff
@@ -260,74 +227,24 @@ def init_fn(scaffold, session, model_portion=None, refresh_base_model=False):
...
-                all_vars, zero_out_adapters = self.subset_to_load(
-                    model_portion, refresh_base_model, all_vars
-                )
-
             global_step_var = tf.train.get_global_step()
 
             for var in all_vars:
-                if global_step_var is not None and global_step_var.name == var.name:
-                    continue
+#                if global_step_var is not None and global_step_var.name == var.name:
```
Comment (madisonmay) [inline finetune/saver.py]: What's the verdict on this bit?

**Review by madisonmay — APPROVED** [review]

File: finetune/saver.py (hunk 7 above)
Comment (benleetownsend) [reply finetune/saver.py]: Oh, this was an accident. I do find myself commenting this out regularly to restart stuff and rarely needing it to be there though. Could instead hide it behind a config argument?

File: finetune/base.py [hunk 8]
```diff
@@ -142,9 +138,17 @@ def _initialize(self):
         self.saver = Saver(
             fallback_filename=self.config.base_model_path,
             exclude_matches=None if self.config.save_adam_vars else "Adam",
-            variable_transforms=[embedding_preprocessor(self.input_pipeline, self.config)],
             save_dtype=self.config.save_dtype,
-            target_model_init_from_base_model=self.config.target_model_init_from_base_model
+        )
+
+    def init_from_checkpoint(self, checkpoint_path):
```
Comment (madisonmay) [inline finetune/base.py:144]: Could we do:

```
def load_from_checkpoint(cls, checkpoint_path):
    model = cls.load(checkpoint_path)
    model.init_from_checkpoint(checkpoint_path)
    return model
```

or am I totally missing how this fns right now?

Comment (benleetownsend) [reply finetune/base.py:144]: checkpoint path is not in the same format as the .save/.load format. we could do something like
```
def load_from_checkpoint(cls, checkpoint_path, **config):
     model = cls(**config)
     model.init_from_checkpoint(checkpoint_path)
     return model
```
